crates/aptos/src/governance/delegation_pool.rs

```diff
@@ -0,0 +1,296 @@
+    proposal_id: Option<u64>,
+    txn_summaries: Vec<TransactionSummary>,
+}
+
+impl SubmitProposalUtils for SubmitProposal {
+    fn get_metadata_url(&self) -> Url {
+        self.metadata_url.clone()
+    }
+
+    #[cfg(feature = "no-upload-proposal")]
+    fn get_metadata_path(&self) -> Option<PathBuf> {
+        self.metadata_path.clone()
+    }
+
+    fn get_compile_proposal_args(&self) -> &CompileScriptFunction {
+        &self.compile_proposal_args
+    }
+
+    fn get_txn_options(&self) -> &TransactionOptions {
+        &self.txn_options
+    }
+}
+
+#[async_trait]
+impl CliCommand<ProposalSubmissionSummary> for SubmitProposal {
+    fn command_name(&self) -> &'static str {
+        "SubmitProposal"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<ProposalSubmissionSummary> {
+        let mut summaries = vec![];
+        if let Some(txn_summary) =
+            delegation_pool_governance_precheck(&self.txn_options, self.delegation_pool_address)
+                .await?
+        {
+            summaries.push(txn_summary);
+        };
+        // Validate the proposal metadata
+        let (script_hash, metadata_hash) = self.compile_proposals().await?;
+        prompt_yes_with_override(
+            "Do you want to submit this proposal?",
+            self.txn_options.prompt_options,
+        )?;
+
+        let txn: Transaction = if self.is_multi_step {
+            self.txn_options
+                .submit_transaction(aptos_stdlib::delegation_pool_create_proposal(
+                    self.delegation_pool_address,
+                    script_hash.to_vec(),
+                    self.metadata_url.to_string().as_bytes().to_vec(),
+                    metadata_hash.to_hex().as_bytes().to_vec(),
+                    true,
+                ))
+                .await?
+        } else {
+            self.txn_options
+                .submit_transaction(aptos_stdlib::delegation_pool_create_proposal(
+                    self.delegation_pool_address,
+                    script_hash.to_vec(),
+                    self.metadata_url.to_string().as_bytes().to_vec(),
+                    metadata_hash.to_hex().as_bytes().to_vec(),
+                    false,
+                ))
+                .await?
+        };
+        let proposal_id = extract_proposal_id(&txn)?;
+        summaries.push(TransactionSummary::from(&txn));
+        Ok(ProposalSubmissionSummary {
+            proposal_id,
+            txn_summaries: summaries,
+        })
+    }
+}
+
+/// Submit a vote on a proposal
+#[derive(Parser)]
+pub struct SubmitVote {
+    /// The address of the delegation pool to vote.
+    #[clap(long)]
+    delegation_pool_address: AccountAddress,
+
+    /// Id of the proposal to vote on
+    #[clap(long)]
+    pub(crate) proposal_id: u64,
+
+    /// Vote to accept the proposal
+    #[clap(long, group = "vote")]
+    pub(crate) yes: bool,
+
+    /// Vote to reject the proposal
+    #[clap(long, group = "vote")]
+    pub(crate) no: bool,
```

Comment:
Note: I think this is fine as long as we take care of a scenario when user input Yes and No at the same time. Which I see you took care of it in `parse_vote_option`